bin/server.js

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env node
+var debug             = require('debug')('github:server');
+var base              = require('taskcluster-base');
+var api               = require('../lib/api');
+var path              = require('path');
+var Promise           = require('promise');
+var exchanges         = require('../lib/exchanges');
+var _                 = require('lodash');
+
+/** Launch server */
+var launch = async function(profile, publisher) {
```

Comment:
Why not just use a different pulse user for testing... presumably `taskcluster-github` is a good pulse username for use in production. But in development you can just use `mrrrgn-debug` and the exchange will be under your username. 

If you create the exchange reference and use it to create `GithubEvents`, you can even use that from the background worker... So you can do an end-to-end test. IMO, setting up a github repo with a `.taskcluster.yml` file is not a bad idea... You can still call the webhook, but otherwise it's very close to an end-to-end test.
In the `.taskcluster.yml` file you can just put `provisionerId: dummy-provisioner` and `workerType: github-dummy-worker` and then inside your test case you can listen for pulse messages on new task created for that provisionerId and workerType. Also your tests will only need credentials to create tasks for the `dummy-provisioner` and `github-dummy-worker`, and it's unlikely to collide with anything people use in production.

Check out `taskcluster-index` here I use the same pattern:
https://github.com/taskcluster/taskcluster-index/blob/8434df52bf1b6a333861da50a4b6c0656e86bfc6/test/index_test.js#L24-L25
Submit an actual task and check that it gets indexed. In your case you don't care about resolution, just that it is submitted.

Well, actually you might later want to take the task through all the stages and validate that background worker synchronizes state with github commit/PR status API (I suspect we want to do that).